### DISCUSSION TITLE
noun_chunks in zh(Chinese)

### DISCUSSION THREAD

**Discussion by JingxinLee** [explosion/spaCy#10695] [Help: Coding & Implementations]
Hi, I am from China and want to train sense2vec from scratch. But I met a problem which is the same as https://github.com/explosion/spaCy/issues/7436#issuecomment-799075740.

What @polm  did was "_copy over the English version, prepare a bunch of example sentences, and then modify the function until the results were correct for all my example sentences. You could use a similar approach for Chinese._"

I am new to Spacy. Could anyone please tell me how to modify the **_noun_chunks_** function?   "use the PhraseMatcher or the DependencyMatcher " to get the labels?

What the **noun_chunks** do is to yield all of the **(index of the noun, index+1, np_label)**, right?

 Thanks in advance.

**Comment by adrianeboyd** [comment]
Paul's description of how to develop a noun chunks method in that comment is still accurate (https://github.com/explosion/spaCy/issues/7436#issuecomment-799075740). Because Chinese uses UD dependency labels, I'd recommend starting with another language as a starting point, one that uses UD in spacy (any language except English or German) and that has a similar noun chunk structure to Chinese.

The method yields tuples that correspond to the noun chunk span, so if you check `doc[start:end]` you'd get the whole span of the noun chunk, like "autonomous cars" for the English example: https://spacy.io/usage/linguistic-features/#noun-chunks

There's one additional step to add the noun chunks method to the language defaults, which looks like this in the language defaults in `__init__py`:

https://github.com/explosion/spaCy/blob/e07500369c654cbd3bf2090838a88a00c600ae2f/spacy/lang/en/__init__.py#L17
